core/src/options/shape.rs

```diff
@@ -72,13 +72,25 @@ impl ToTokens for Shape {
         } else {
             let en = &self.enum_values;
             let st = &self.struct_values;
+
+            let check_empty_enum = if en.supports_none() {
+                let ty = en.prefix.trim_end_matches('_');
+                quote!(return ::darling::export::Err(::darling::Error::unsupported_shape(#ty));)
+            } else {
+                quote!()
+            };
+
             quote! {
                 match *__body {
                     ::syn::Data::Enum(ref data) => {
                         fn validate_variant(data: &::syn::Fields) -> ::darling::Result<()> {
                             #en
                         }
 
+                        if data.variants.is_empty() {
+                            #check_empty_enum
+                        }
+
```

Comment:
If the receiver doesn't support any enum shapes, should we surface one diagnostic rather than one per variant?

Reply:
Yeah, would be better to surface one diagnostic.

Reply:
Updated!